### PR TITLE
Initial aircraft tracker subsystem implementation

### PR DESCRIPTION
**Pull request by StevenFrost**
This is a super basic implementation of the aircraft tracker subsystem, which fires an event to the internal dispatcher when the user aircraft changes position.

### CODE REVIEW

File: Source/SimCacheModule/Source/Subsystems/AircraftTracker/AircraftTracker.cpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	{
+		SimConnectClient.UnregisterUserAircraftPositionListener( UserAircraftPositionUpdatedHandle );
+	}
+}
+
+// -----------------------------------------------------------------------------
+
+void AircraftTracker::OnAircraftPositionUpdated( const Utils::EarthCoordinate CurrentPosition )
+{
+	InternalEventDispatcher.FireEvent( AircraftPositionUpdatedEvent() );
```
Comment (ollyau) [inline Source/SimCacheModule/Source/Subsystems/AircraftTracker/AircraftTracker.cpp]: Is this supposed to use `CurrentPosition`?

**Review by ollyau — CHANGES_REQUESTED** [review]
As discussed via Discord, I'm not sure this works currently? :P

File: Source/SimCacheModule/Source/Subsystems/AircraftTracker/AircraftTracker.cpp (hunk 1 above)
Comment (StevenFrost) [reply Source/SimCacheModule/Source/Subsystems/AircraftTracker/AircraftTracker.cpp]: Yep! Should be all hooked up now.

File: Source/SimCacheModule/Source/Subsystems/AircraftTracker/AircraftTracker.cpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	{
+		SimConnectClient.UnregisterUserAircraftPositionListener( UserAircraftPositionUpdatedHandle );
+	}
+}
+
+// -----------------------------------------------------------------------------
+
+void AircraftTracker::OnAircraftPositionUpdated( const Utils::EarthCoordinate CurrentPosition )
+{
+	InternalEventDispatcher.FireEvent( AircraftPositionUpdatedEvent( CurrentPosition ) );
```
Comment (ollyau) [inline Source/SimCacheModule/Source/Subsystems/AircraftTracker/AircraftTracker.cpp:80]: Do we want a todo item here mentioning that once we have caching, we may want to make firing the event conditional? (e.g. if the change in position is greater than some epsilon, say, idk, a centimeter?  or maybe if the new `CurrentPosition` simply doesn't equal the cached one?)

File: Source/SimCacheModule/Source/Subsystems/AircraftTracker/AircraftTracker.cpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	, SimConnectClient( SimConnectClient )
+	, UserAircraftPositionUpdatedHandle()
+{
+}
+
+// -----------------------------------------------------------------------------
+
+bool AircraftTracker::Initialize()
+{
+	// TODO: Only bind the position updated event when we have a tracked cache
```
Comment (ollyau) [inline Source/SimCacheModule/Source/Subsystems/AircraftTracker/AircraftTracker.cpp:34]: Regarding this todo note -- are we ever going to want to track the position of the user aircraft / ownship outside having a tracked cache?  (e.g. maybe on the cache selection frontend / FS2024 EFB app, we might want to be able to get the current location to filter the list by some form of proximity, so it doesn't suggest caches on the other side of the globe?)

**Review by ollyau — APPROVED** [review]
I think this looks good enough to work towards the first playable! :)

**Pull request merged**